integration/test/counter_test.go

```diff
@@ -0,0 +1,184 @@
+package main
+
+import (
+	"fmt"
+	"github.com/Netapp/harvest-automation/test/installer"
+	"github.com/Netapp/harvest-automation/test/utils"
+	"github.com/hashicorp/go-version"
+	"github.com/netapp/harvest/v2/cmd/collectors"
+	rest2 "github.com/netapp/harvest/v2/cmd/tools/rest"
+	"github.com/netapp/harvest/v2/pkg/auth"
+	"github.com/netapp/harvest/v2/pkg/conf"
+	"github.com/netapp/harvest/v2/pkg/logging"
+	"github.com/netapp/harvest/v2/pkg/tree"
+	"github.com/netapp/harvest/v2/pkg/tree/node"
+	"github.com/netapp/harvest/v2/pkg/util"
+	"github.com/rs/zerolog/log"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+)
+
+type counterData struct {
+	api          string
+	restCounters []string
+	perfCounters []string
+}
+
+var replacer = strings.NewReplacer("{", "", "}", "", "^^", "", "^", "")
+
+func TestCounters(t *testing.T) {
+	var (
+		poller *conf.Poller
+		client *rest2.Client
+	)
+
+	utils.SkipIfMissing(t, utils.Regression)
+	_, err := conf.LoadHarvestConfig(installer.HarvestConfigFile)
+	if err != nil {
+		log.Fatal().Err(err).Msg("Unable to load harvest config")
+	}
+
+	pollerName := "dc1"
+	if poller, err = conf.PollerNamed(pollerName); err != nil {
+		log.Fatal().Err(err).Str("poller", pollerName).Msgf("")
+	}
+	if poller.Addr == "" {
+		log.Fatal().Str("poller", pollerName).Msg("Address is empty")
+	}
+	timeout, _ := time.ParseDuration(rest2.DefaultTimeout)
+
+	if client, err = rest2.New(poller, timeout, auth.NewCredentials(poller, logging.Get())); err != nil {
+		log.Fatal().Err(err).Str("poller", pollerName).Msg("error creating new client")
+		os.Exit(1)
+	}
+
+	if err = client.Init(5); err != nil {
+		log.Fatal().Err(err).Msg("client init failed")
+	}
+
+	restCounters := processRestCounters(client)
+	if err = invokeRestCall(client, restCounters); err != nil {
+		log.Error().Err(err).Msg("rest call failed")
+	}
+
+}
+
+func invokeRestCall(client *rest2.Client, counters map[string][]counterData) error {
+	for _, countersDetail := range counters {
+		for _, counterDetail := range countersDetail {
+			href := rest2.NewHrefBuilder().
+				APIPath(counterDetail.api).
+				Fields(counterDetail.restCounters).
+				CounterSchema(counterDetail.perfCounters).
+				Build()
+
+			if _, err := collectors.InvokeRestCall(client, href, logging.Get()); err != nil {
+				log.Info().Msgf("%s", href)
```

Comment:
Might be better to augment the error with the href like so and then remove the log.Info
`return fmt.Errorf("failed to invoke rest href=%s call: %w", href, err)`